File: app/src/main/java/com/androidwind/androidquick/sample/mvvm/livedata/LiveDataViewModel.java
package com.androidwind.androidquick.sample.mvvm.livedata;

import com.androidwind.androidquick.mvvm.livedata.EventMutableLiveData;

import androidx.lifecycle.ViewModel;

/**
 * @author ddnosh
 * @website http://blog.csdn.net/ddnosh
 */
public class LiveDataViewModel extends ViewModel {
    private LiveDataRepository repository;
    private EventMutableLiveData<String> mLiveData;

    public EventMutableLiveData<String> getLiveData() {
        return mLiveData;
    }

    public LiveDataViewModel() {
        repository = new LiveDataRepository();
        mLiveData = repository.getLiveData();
    }

    public void test() {
        repository.test();
    }
}

File: app/src/main/java/com/github/ddnosh/arabbit/sample/livedata/LiveDataViewModel.kt
package com.github.ddnosh.arabbit.sample.livedata

import androidx.lifecycle.ViewModel
import com.github.ddnosh.arabbit.mvvm.livedata.EventMutableLiveData

/**
 * @author ddnosh
 * @website http://blog.csdn.net/ddnosh
 */
class LiveDataViewModel : ViewModel() {
    private val repository: LiveDataRepository = LiveDataRepository()
    val liveData: EventMutableLiveData<String> = repository.liveData
    fun test() {
        repository.test()
    }
}